### PR TITLE
Remove unnecessary auto-boxing.

### PR DESCRIPTION
**Pull request by ZahidFKhan**

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java [hunk 1]
```diff
@@ -85,8 +85,8 @@ protected UnicastingDispatcher getDispatcher() {
 	@Override
 	protected void onInit() {
 		super.onInit();
-		if (this.maxSubscribers == null) {
-			Integer max = getIntegrationProperty(IntegrationProperties.CHANNELS_MAX_UNICAST_SUBSCRIBERS, Integer.class);
+		if (this.maxSubscribers == 0) {
+			int max = getIntegrationProperty(IntegrationProperties.CHANNELS_MAX_UNICAST_SUBSCRIBERS, int.class);
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java:89]: Sorry, but this is not correct.
See the behavior:
```
protected <T> T getIntegrationProperty(String key, Class<T> tClass) {
		return this.defaultConversionService.convert(this.integrationProperties.getProperty(key), tClass);
	}
```

Now imagine that `integrationProperties.getProperty()` returns `null`.
That `ConversionService` will try to convert our `null` to the expected type - `int.class` in our case.
And we end up with this:
```
private Object handleResult(@Nullable TypeDescriptor sourceType, TypeDescriptor targetType, @Nullable Object result) {
		if (result == null) {
			assertNotPrimitiveTargetType(sourceType, targetType);
		}
		return result;
	}
...
	private void assertNotPrimitiveTargetType(@Nullable TypeDescriptor sourceType, TypeDescriptor targetType) {
		if (targetType.isPrimitive()) {
			throw new ConversionFailedException(sourceType, targetType, null,
					new IllegalArgumentException("A null value cannot be assigned to a primitive type"));
		}
	}
```

so, what is the motivation for this change if it is not going to work for every use-case?

Thanks for any explanations!

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java (hunk 1 above)
Comment (ZFK07) [reply spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java]: Hi @artembilan 
Yes, that's correct, I agree the conversion will fail and will throw an exception. 🙈 
I have reverted back to the previous code.

However I believe unnecessary auto unboxing is not needed:

	
	public void setMaxSubscribers(int maxSubscribers) {
		..
	}
	
and needs to be changed to 
	
	public void setMaxSubscribers(Integer maxSubscribers) {
		..
	}

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public void setFailover(boolean failover) {
 	 * channel's dispatcher.
 	 * @param maxSubscribers The maximum number of subscribers allowed.
 	 */
-	public void setMaxSubscribers(int maxSubscribers) {
+	public void setMaxSubscribers(Integer maxSubscribers) {
```
[the remark this answers is not in the document]
Comment (ZFK07) [reply spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java:75]: Hi @artembilan 
Do you think having a cross-check:

    Integer max = getIntegrationProperty(IntegrationProperties.CHANNELS_MAX_UNICAST_SUBSCRIBERS, Integer.class);
       
	if (!Objects.isNull(max) && max>0) {
	    setMaxSubscribers(max);	
	}
	
would that be correct?

[the remark this answers is not in the document]
Comment (ZFK07) [reply spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java:75]: @artembilan I am new to Spring Integration and out of curiosity want to know the reason why are we allowing `CHANNELS_MAX_UNICAST_SUBSCRIBERS` to be negative? And what happens if the property is set to be negative?

[the remark this answers is not in the document]
Comment (artembilan) [reply spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java]: The property is used in the `AbstractDispatcher`:
```
public synchronized boolean addHandler(MessageHandler handler) {
		Assert.notNull(handler, "handler must not be null");
		Assert.isTrue(this.handlers.size() < this.maxSubscribers, "Maximum subscribers exceeded");
```

So, if we set that property to `0` or negative, that would effectively mean we don't allow subscription to this channel.
Very unlikely that anyone would do this in the real application, but this is only a reason I can explain why we allow a non-positive number of subscribers. 😄

**Comment by artembilan** [comment]
@ZFK07 ,

Thank you very much for the contribution; looking forward for more!

File: spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZFK07) [reply spring-integration-core/src/main/java/org/springframework/integration/channel/DirectChannel.java]: Thanks, @artembilan for the explanation. I understood the reason behind it.
